packages/elastic-apm-synthtrace/src/lib/interval.ts

```diff
@@ -6,27 +6,69 @@
  * Side Public License, v 1.
  */
 import moment from 'moment';
+import { ApmFields } from './apm/apm_fields';
+import { SpanIterable } from './span_iterable';
+import { SpanGenerator } from './span_generator';
 
-export class Interval {
+export function parseInterval(interval: string): [number, string] {
+  const args = interval.match(/(\d+)(s|m|h|d)/);
+  if (!args || args.length < 3) {
+    throw new Error('Failed to parse interval');
+  }
+  return [Number(args[1]), args[2] as any];
+}
+
+export class Interval implements Iterable<number> {
   constructor(
-    private readonly from: number,
-    private readonly to: number,
-    private readonly interval: string
-  ) {}
+    public readonly from: Date,
+    public readonly to: Date,
+    public readonly interval: string,
+    public readonly yieldRate: number = 1
+  ) {
+    [this.intervalAmount, this.intervalUnit] = parseInterval(interval);
```

Comment:
I think it's a little easier for the reader if we don't directly assign to class properties (e.g. by using an intermediate variable). I think most folks are not used to this approach.

Reply:
I wanted to localize the parsing of `amount` and `unit` to the constructor since its immutable state that otherwise `_generate()` needs to calculate every time a new enumeration is started.